src/ai/ai.cpp

```diff
@@ -270,6 +279,11 @@ bool ai_will_accept_alliance(sys::state& state, dcon::nation_id target, dcon::na
 	if(!ai_is_close_enough(state, target, from))
 		return false;
 
+	//We'd rather want to kill them rather than ally
+	if(military::can_use_cb_against(state, from, target)) {
+		return false;
```

Comment:
This will need a corresponding line added to the offer alliance button in the diplomacy window explaining why the ai won't accept